modules/rntuple.mjs

```diff
@@ -148,7 +149,7 @@ deserializeFooter(footer_blob) {
     // Feature flag(32 bits)
     this._readFeatureFlags(reader);
     // Header checksum (64-bit xxhash3)
-    this.headerChecksum = reader.readU64(); 
+    this.headerChecksumFromFooter = reader.readU64(); 
```

Comment:
The check on `headerChecksumFromFooter === headerEnvelopeChecksum` should happen here rather than in the test, as if it fails the RNTuple is corrupted and the reader should stop and report error.

Reply:
(also, this can be a local variable)